packages/block-editor/src/utils/dom.js

```diff
@@ -86,3 +86,24 @@ export function isInsideRootBlock( blockElement, element ) {
 export function hasInnerBlocksContext( element ) {
 	return !! element.querySelector( '.block-editor-block-list__layout' );
 }
+
+/**
+ * Finds the block client ID given any DOM node inside the block.
+ *
+ * @param {Node} node DOM node.
+ *
+ * @return {?string} Client ID or undefined if the node is not part of a block.
```

Comment:
It's an error we've been making for a while now, but `?` indicates a value which can explicitly be _nullable_, not undefined.

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#nullable-undefined-and-void-types

This would be more correct as:

```suggestion
 * @return {string|undefined} Client ID or undefined if the node is not part of a block.
```